### ISSUE TITLE
Missing permission as superuser/admin

### ISSUE THREAD

**Issue by arnoldnipper** [peeringdb/peeringdb#735]
When deleting a user I get the following result.

![20200602 PeeringDB 00](https://user-images.githubusercontent.com/16888757/83573657-308e5600-a52c-11ea-8a73-a897f43475bf.png)

@peeringdb/ac, can you please confirm. User is "paulo.dias"

As superuser/admin, I should be able to delete anything. Or at least get good advice on how to resolve the conflict

**Comment by arnoldnipper** [comment]
@vegu what prevents a superuser from deleting objects?

**Comment by grizz** [comment]
@arnoldnipper this is the cannot delete if an object has child deps that the DoTF put in. I agree it should list the linked objects, as in this case, it would be acceptable to cascade the delete.

**Comment by arnoldnipper** [comment]
> @arnoldnipper this is the cannot delete if an object has child deps that the DoTF put in. I agree it should list the linked objects, as in this case, it would be acceptable to cascade the delete.

I understand, @grizz. But shouldn't the superuser be able to delete anything?

**Comment by arnoldnipper** [comment]
With release 2.22.0 #696 was implemented which blocks deletion of some objects if they still are referenced. However, the superuser (== @peeringdb/ac ) should be able to to do a _cascaded_ delete. I.e. if a superuser tried to delete such an object, first all references will be deleted and then the object itself.

@pc, please vote

**Comment by alexfguimaraes** [comment]
Arnold,

It’s a very import resource and how to use use that should be very wise. With 2 or 3 “Are you sure?” or a view that show the chain of deletion.

Yes

Att
Alexandre

**Comment by shane-kerr** [comment]
+1

**Comment by grizz** [comment]
> > @arnoldnipper this is the cannot delete if an object has child deps that the DoTF put in. I agree it should list the linked objects, as in this case, it would be acceptable to cascade the delete.
> 
> I understand, @grizz. But shouldn't the superuser be able to delete anything?

No, no user should be able to cascade delete a tree of active objects by deleting one object.

In this case, a user org request should absolutely be cascade deleted with the user. I'm sure there's a handful of other objects that would fall into that category as well.

**Comment by arnoldnipper** [comment]
> No, no user should be able to cascade delete a tree of active objects by deleting one object.

@grizz, so you even want @peeringdb/ac to delete every single object instead of simply deleting the referenced object? E.g. there is a facility which went out of operation but still has 12 networks listed.

**Comment by grizz** [comment]
Verses accidentally deleting something that has 12 active things listed under it? I think so. Neither is ideal, but it also doesn't happen all that often so I would lean towards manually deleting stale entries.

I'm also open to a third idea of doing it.

**Comment by arnoldnipper** [comment]
> Verses accidentally deleting something that has 12 active things listed under it? I think so.

Come on ... are you forcing me to write a little script what exactly does that? That even works w/o having to ACK anything. And accidentally via the Admin GUI? Of course, may happen. But admins also always have to confirm if they delete a record.

Convinced, @grizz?

**Comment by arnoldnipper** [comment]
@peeringdb/pc, @peeringdb/ac would like to be able to cascade-delete objects via the AC GUI. Why? If e.g. an IX or Facility goes out of operation and should be deleted, only AC can do when theses objects still have children. 

Currently, AC has to delete each of the children first before the parent object can be deleted. Hence, a cascade-delete GUI feature would come in handy.

@grizz objects that with this tool AC could accidentally delete something that has a lot of things listed under it. Fair point. However, it only takes me 10 minutes to write a script which does exactly this. So, what is the point not to provide such a tool within the GUI?

@peeringdb/ac and @peeringdb/pc, please discuss!

**Comment by DarwinCosta** [comment]
> On 7 Dec 2020, at 07:13, Arnold Nipper <notifications@github.com> wrote:
> 
> 
> @peeringdb/pc <https://github.com/orgs/peeringdb/teams/pc>, @peeringdb/ac <https://github.com/orgs/peeringdb/teams/ac> would like to be able to cascade-delete objects via the AC GUI. Why? If e.g. an IX or Facility goes out of operation and should be deleted, only AC can do when theses objects still have children.
> 
> Currently, AC has to delete each of the children first before the parent object can be deleted. Hence, a cascade-delete GUI feature would come in handy.
> 
> 
For operational proposes - this is indeed an handy feature to have it on the GUI. Specially when we are talking about an IX or Facility with a lot children where we don’t have to delete it one by one.
> @grizz <https://github.com/grizz> objects that with this tool AC could accidentally delete something that has a lot of things listed under it. Fair point. However, it only takes me 10 minutes to write a script which does exactly this. So, what is the point not to provide such a tool within the GUI?
> 
> 
Assuming that an accident might happen, how quick or would it be possible to easily revert the situation?
> @peeringdb/ac <https://github.com/orgs/peeringdb/teams/ac> and @peeringdb/pc <https://github.com/orgs/peeringdb/teams/pc>, please discuss!
> 
> 
Cheers,
Darwin-.

**Comment by arnoldnipper** [comment]
> Assuming that an accident might happen, how quick or would it be possible to easily revert the situation?
>

IMHO that's not the point. If there is no GUI tool, I will write a little script which does the trick. So, the question is: why not providing a GUI feature when AC can do this via own script anyway?

**Comment by DarwinCosta** [comment]
> On 7 Dec 2020, at 16:11, Arnold Nipper <notifications@github.com> wrote:
> 
> 
> Assuming that an accident might happen, how quick or would it be possible to easily revert the situation?
> 
> IMHO that's not the point. If there is no GUI tool, I will write a little script which does the trick. So, the question is: why not providing a GUI feature when AC can do this via own script anyway?
> 
> 
Got it.

Well I think we all can run a script to do the work - but having the feature in the AC GUI is way better as then everything is logged and can easily be tracked. My 2cts.

Wondering what the rest of AC team thinks about it?

Cheers,

Darwin-.

**Comment by peterhelmenstine** [comment]
Prefer a GUI

$0.02

**Comment by grizz** [comment]
> Convinced, @grizz?

Not entirely. How often is that feature even needed?

**Comment by arnoldnipper** [comment]
> Not entirely. How often is that feature even needed?

Every time an object with children is deleted. This happens roughly once per month.

**Comment by alexfguimaraes** [comment]
+1

**Comment by arnoldnipper** [comment]
@peeringdb/ac unanimously wants this feature. We should trust them that they know what they do. @peeringdb/pc please ACK.

**Comment by job** [comment]
+1

**Comment by ynbrthr** [comment]
+1

**Comment by vegu** [comment]
**summary**
- admincom should be able to cascade delete anything through the django admin ui
